subprojects/model-core/src/main/java/org/gradle/api/internal/provider/ValueSupplier.java

```diff
@@ -254,24 +362,56 @@ public boolean isMissing() {
 
         @Override
         public T get() throws IllegalStateException {
+            runSideEffect();
             return result;
         }
 
         @Override
         public T orNull() {
+            runSideEffect();
```

Comment:
❓ Why not delegate to `get()` here? It would be nice to limit the call sites to `runSideEffect` IMO.

Reply:
Done